integration-templates/zoho-crm/zoho-crm-accounts.ts

```diff
@@ -0,0 +1,124 @@
+import type { ZohoCRMAccount, NangoSync } from './models';
+
+export default async function fetchData(nango: NangoSync) {
+    try {
+        const responses = await getAllAccounts(nango, '/crm/v2/Accounts');
+        if (responses.length > 0) {
+            const mappedAccounts: ZohoCRMAccount[] = responses.map(mapAccounts);
+            // Save Accounts
+            await nango.batchSave(mappedAccounts, 'ZohoCRMAccount');
+        } else {
+            await nango.log('No Accounts found.');
+        }
+    } catch (error: any) {
+        await nango.log(`Error in fetchData: ${error.message}`);
+    }
+}
+
+async function getAllAccounts(nango: NangoSync, endpoint: string): Promise<any[]> {
+    const accounts: any[] = [];
+    let page = 1;
+    const perPage = 100;
+    let response: { data: { data?: any[]; info?: { more_records?: boolean }; nextPage?: string } };
+
+    try {
+        do {
+            const { params, headers } = generateApiRequestParams(nango, page, perPage);
+
+            response = await nango.get({
```

Comment:
To simplify this you could look into using Nango's built-in pagination method. It supports page + offset pagination. A good example is in the confluence pages sync.
If the pagination method is the same for most of the endpoints of the Zoho CRM API you can define the config in the `providers.yaml` for it.